Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,235 @@
+            IndexerRunningException {
+        ConceptEntry entry = conceptManager.getConceptEntry(conceptId);
+        // Creating concept wrapper with all the values, because in future we
+        // will be including manipulations on deleted wrappers as well.
+        getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+        return conceptManager.addConceptListEntry(entry, userName);
+    }
+
+    public void getConceptEntryFromConceptMergeBean(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        entry.setPos(conceptMergeBean.getSelectedPosValue());
+        entry.setConceptList(conceptMergeBean.getSelectedListName());
+        entry.setTypeId(conceptMergeBean.getSelectedTypeId());
+
+        String prefix = ",";
+        String words = conceptMergeBean.getWords().stream().map(i -> i.toString()).collect(Collectors.joining(prefix));
+        entry.setWord(words);
+
+        String description = conceptMergeBean.getDescriptions().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setDescription(description);
+
+        String synonymIds = conceptMergeBean.getSynonymsids().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSynonymIds(synonymIds);
+
+        String equals = conceptMergeBean.getEqualsValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setEqualTo(equals);
+
+        String similar = conceptMergeBean.getSimilarValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSimilarTo(similar);
+
+        String wordnetIds = conceptMergeBean.getWordnetIds().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setWordnetId(wordnetIds);
+
+        addAlternativeIdsToEntry(entry, conceptMergeBean);
+        addMergedIdsToEntry(entry, conceptMergeBean);
+    }
+
+    private void addAlternativeIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        List<String> conceptIds = conceptMergeBean.getConceptIds();
+        String selectedId = conceptMergeBean.getSelectedConceptId();
+        Set<String> alternativeIds = new HashSet<>();
+
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(selectedId) || ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                    .getConceptTypeByConceptId(selectedId)) {
+                alternativeIds.add(conceptId);
+                ConceptEntry conceptEntry = conceptManager.getConceptEntry(conceptId);
+                alternativeIds.addAll(conceptEntry.getAlternativeIds());
+            }
+        }
+        entry.setAlternativeIds(alternativeIds);
+    }
+
+    private Predicate<String> isSpecificWordnetConcept(String selectedId) {
+        return p -> !p.equalsIgnoreCase(selectedId)
```

Comment:
I would use a more descriptive variable name here. Since ```p``` will always refer to an id, and ```id``` is not much longer than ```p```, you should use that here. It is not clear, why you would call the variable ```p``` here but you code would be much more understandable with a proper variable name.